tests/data_context/cloud_data_context/test_datasource_crud.py

```diff
@@ -358,3 +369,281 @@ def test_cloud_data_context_add_datasource(
+
+    # 3. Make sure no exceptions are raised when saving.
+    with patch(
+        "great_expectations.data_context.store.datasource_store.DatasourceStore.set",
+        autospec=True,
+        return_value=datasource_config_with_names_modified,
+    ):
+
+        updated_datasource: Union[
+            LegacyDatasource, BaseDatasource
+        ] = context.save_datasource(new_datasource)
+
+        # Make sure the datasource config got into the context config
+        assert len(context.config.datasources) == 1
+        assert (
+            context.config.datasources[datasource_name]
+            == datasource_config_with_names_modified
+        )
+
+        cached_datasource = context._cached_datasources[datasource_name]
+
+        # Make sure the datasource did not get duplicated in the cache
+        assert len(context._cached_datasources) == 1
+
+        # Make sure the name was updated
+        updated_datasource_data_connector_name: str = tuple(
+            updated_datasource.data_connectors.keys()
+        )[0]
+        assert updated_datasource_data_connector_name == new_data_connector_name
+
+        # Make sure the stored and returned datasource is the same one as the cached datasource
+        assert id(updated_datasource) == id(cached_datasource)
+        assert updated_datasource == cached_datasource
+
+        # Make sure the stored and returned datasource is a different instance
+        assert not id(cached_datasource) == id(new_datasource)
+        assert not id(updated_datasource) == id(new_datasource)
+        assert not cached_datasource == new_datasource
+        assert not updated_datasource == new_datasource
+
+        # Make sure the stored and returned datasource are otherwise equal
+        serializer: AbstractConfigSerializer = DictConfigSerializer(
+            schema=datasourceConfigSchema
+        )
+        updated_datasource_dict = serializer.serialize(
+            datasourceConfigSchema.load(updated_datasource.config)
+        )
+        orig_datasource_dict = serializer.serialize(
+            datasourceConfigSchema.load(new_datasource.config)
+        )
+
+        for attribute in ("name", "execution_engine", "data_connectors"):
+            assert updated_datasource_dict[attribute] == orig_datasource_dict[attribute]
+
+
+@pytest.mark.parametrize(
+    "data_context_fixture_name,data_context_type",
+    [
+        pytest.param(
+            "empty_base_data_context_in_cloud_mode",
+            BaseDataContext,
+            id="BaseDataContext",
+        ),
+        pytest.param(
+            "empty_data_context_in_cloud_mode",
+            DataContext,
+            id="DataContext",
+        ),
+        pytest.param(
+            "empty_cloud_data_context",
+            CloudDataContext,
+            id="CloudDataContext",
+        ),
+    ],
+)
+@pytest.mark.cloud
+@pytest.mark.unit
+def test_cloud_backed_data_context_save_datasource_empty_store(
```

Comment:
Please limit this test to the stated description: specifically, it should cover the code path that exercises cloud mode (and not the others)